typespec-tests/src/main/java/com/type/union/UnionClient.java

```diff
@@ -42,7 +46,7 @@ public final class UnionClient {
      * </p>
      * <pre>{@code
      * {
-     *     simpleUnion: ModelWithSimpleUnionPropertySimpleUnionModelBase (Required)
+     *     simpleUnion: BinaryData (Required)
```

Comment:
javadoc will be addressed in future

We hope it would let use know which type can be assigned to this `BinaryData`